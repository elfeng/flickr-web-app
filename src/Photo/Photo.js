import React from 'react';
import './Photo.css';

export default function Photo(props) {
    return (
        <div className="photo">
            <a href={props.linkUrl}>
                <img src={props.snapshotUrl} alt={props.title}/>
                <div>
                    {props.dateTakenFormatted}
                </div>
            </a>
        </div>
    );

};
