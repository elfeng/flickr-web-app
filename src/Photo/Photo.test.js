import React from 'react';
//import ReactTestUtils from 'react-dom/test-utils';
//import ReactTestUtils from 'react-addons-test-utils';
import renderer from 'react-test-renderer';
import Photo from './Photo';

function createNodeMock(element) {
  if (element.type === 'div') {
    return {
      focus() {},
    };
  }
  return null;
}

test('renders without crashing', () => {
  const tree = renderer.create(
    <Photo />,
    {createNodeMock}
  ).toJSON();
  // const testUtilsRendered = ReactTestUtils.createRenderer();
  // testUtilsRendered.render(<Photo />);
  // const result = testUtilsRendered.getRenderOutput();
  expect(tree.type).toBe('div');
});

test('should match the snapshot', () => {
  const component = renderer.create(
    <Photo
      linkUrl="https://www.flickr.com/photos/21668212@N02/25277702294"
      snapshotUrl="https://farm2.staticflickr.com/1658/25277702294_3514fc66de.jpg"
      title="alpagas at sajama"
      dateTakenFormatted="April 1, 2015"/>
      );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
