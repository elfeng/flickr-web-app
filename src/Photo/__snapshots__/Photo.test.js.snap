// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div
  className="photo"
>
  <a
    href="https://www.flickr.com/photos/21668212@N02/25277702294"
  >
    <img
      alt="alpagas at sajama"
      src="https://farm2.staticflickr.com/1658/25277702294_3514fc66de.jpg"
    />
    <div>
      April 1, 2015
    </div>
  </a>
</div>
`;
